subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,113 @@
+project(
+  'speexdsp',
+  'c',
+  version : '1.2.0',
+  license : 'bsd-like',
+  default_options : [
+    'c_std=c99',
+    'buildtype=release',
+    'default_library=both',
+    'b_lto=true',
+    'warning_level=0'])
+
+# library versioning from speexdsp-1.2.0's configure.ac
+# SPEEXDSP_LT_CURRENT=6, SPEEXDSP_LT_REVISION=1, SPEEXDSP_LT_AGE=5
+lib_current = 6
+lib_revision = 1
+lib_age = 5
+lib_soversion = lib_current - lib_age
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+cdata = configuration_data()
+
+# Set SIMD config data and flags
+simd_type = 'None'
+if host_cpu_family.startswith('x86')
+  cdata.set('USE_SSE', true)
+  cdata.set('USE_SSE2', true)
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  simd_type = 'SSE2'
+
+elif host_cpu_family == 'arm'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  simd_type = 'NEON (vfpv4)'
+
+elif host_cpu_family == 'aarch64'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  simd_type = 'NEON (armv8)'
+endif
+
+# Set numerical representation config data
+numerical_type = ''
+fft_library = ''
+if get_option('use-fixed-point')
+  # the fixed-point implementation doesn't use SSE intrinsics (and arch.h
+  # checks for this), however we still let the compiler pack floats into
+  # SIMD calls instructions whenever its cost model permits. 
+  cdata.set('USE_SSE', false)
```

Comment:
This seems to be changing the `USE_SSE` coniguration data variable but the `summary()` call will still report the use of SSE2, unless I am missing something. Is this intended?

Reply:
It's intended; I've tried to describe this better in the comment:

``` text
  # The fixed-point implementation isn't written using SSE intrinsics
  # and "arch.h" checks that USE_SSE is disabled when using fixed-point,
  # so we explicitly disable it here to pass that check:
  cdata.set('USE_SSE', false)
  # However, we still let the compiler pack integer operations into SIMD
  # calls whenever possible and we inform the user that SIMD is active in
  # the Summary block.
```